sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsError.cs

```diff
@@ -1,20 +1,58 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Azure.AI.TextAnalytics.Models;
+using Azure.Core;
+
 namespace Azure.AI.TextAnalytics
 {
     /// <summary>
     /// Text Analytics Error.
     /// </summary>
-    public struct TextAnalyticsError
+    [CodeGenModel("TextAnalyticsError")]
+    public partial struct TextAnalyticsError
     {
         internal TextAnalyticsError(string code, string message, string target = null)
+            : this()
         {
+            if (code == null)
+            {
+                throw new ArgumentNullException(nameof(code));
+            }
+            if (message == null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
             ErrorCode = code;
             Message = message;
             Target = target;
         }
 
+        internal TextAnalyticsError(ErrorCodeValue code, string message, string target, InnerError innererror, IEnumerable<TextAnalyticsError> details)
```

Comment:
Although we don't use this constructor, codegen generates it, so putting it here so we can assign value to the other properties.
@pakrym is there a better way to do this?

Reply:
What do you mean other properties?